### PR TITLE
Best Practices for Accessible Web Content

### PR DESCRIPTION
**Pull request by hxmattobee**
This introduces a set of best practices for developing accessible web content with HTML, CSS and JavaScript. It extends and complements the best practices already described for those technologies in their own documents.

### CODE REVIEW

**Comment by elliottcrush** [comment]
Taking review.

**Comment by elliottcrush** [comment]
Really good @hxmattobee. Lots of great content in here :+1:

File: accessibility-best-practices.md [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+    <label for="year" class="hidden">Year</label>
+    <input id="year" type="text">
+</fieldset>
+```
+
+### Labels for checkboxes and radio buttons
+As well as being explicitly associated with `for` and `id` attributes, labels for checkboxes and radio buttons should appear *after* the control in the markup and be displayed *to the right* of the control on screen. This convention ensures compatibility with the widest range of assistive technologies, and is also consisitent with UI patterns established by Microsoft and Apple.
+
+### Identifying mandatory fields
+A mandatory field should be identified in its label, in a way that doesn't rely on vision, so that it can be announced by assistive technology when the field receives focus. This could be an asterisk (*) or simply say "required". Where an asterisk or other symbol is used, this should be explained at the top of the form (e.g. "fields marked with * are required.") Where all fields in a form are mandatory, it is acceptable to simply say "all fields are mandatory", but probably easier (technically) to adopt the same approach throughout and mark all individual fields.
```
Comment (msaspence) [inline accessibility-best-practices.md]: Mentioning the use of `<abbr title='required field'>*</abbr>` in required labels might be a nice touch here.

Comment (hxmattobee) [inline accessibility-best-practices.md]: Good call @msaspence. Added.

**Comment by jackdclark** [comment]
Reviewing :mag:

File: accessibility-best-practices.md [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+```html
+<label for="location">Flying from?</label>
+<select name="location" id="location">
+    <option value="ABZ">Aberdeen</option>
+    <option value="BFS">Belfast International</option>
+    ...
+</select>
+```
+
+Placeholder text (a default vaule in the field) should **not** be used instead of a separate `label` element. Placeholders are cleared on focus and often not announced at all by screen readers or other assistive technology.
```
Comment (jackdclark) [inline accessibility-best-practices.md]: ~~vaule~~**value**

**Comment by jackdclark** [comment]
This is great @hxmattobee, nice work! :+1:

**Comment by Joezo** [comment]
Really good @hxmattobee :D

**Pull request merged**